### PR TITLE
Feature/165 return content

### PR DESCRIPTION
**Pull request by jjosef**
I've added this as a new feature for #165

### CODE REVIEW

**Comment by Robert-W** [comment]
First peek, this looks pretty straight forward to me. Just as an FYI, we generate all the code in the profiles, so once this is ready and approved (ideally before we merge), we will need to make the update in our generator so that these updates do not get overridden next time we generate code for an update. Also, in our next major release, we removed the profiles all together and put response utils in a separate package, so I may need to think of another way to implement this in the [2.0 branch](https://github.com/Asymmetrik/node-fhir-server-core/tree/feat/version_2_0_0)

**Comment by jjosef** [comment]
Thank you Robert, is there anything I can help with there? Is the generator in a different project? I did not see it in the source or README

**Comment by Robert-W** [comment]
Unfortunately not at the moment, it is a separate project and it has not been open sourced yet since we are still not 100% what we are doing with it. On top of using it for this and our [graphql-fhir](https://github.com/Asymmetrik/graphql-fhir) project, we also use it internally for our enterprise level software. So the future of that tool is still up in the air. 

So at the moment, we essentially just add it as a task to our sprint board and push out new versions as soon as it's ready. With 2.0, the utility is in a separate project that is open sourced, so we can probably add something for it there. That project lives at https://github.com/Asymmetrik/phx-tools, specifically the `fhir-response-util` package.

**Comment by Robert-W** [comment]
Hey @jjosef, yea that sounds like a pretty good idea to me. I am in the process of writing the migration guide for 2.0.0 and getting the remaining features put in, so I apologize for the delays on getting this merged. The goal is to get all this taken care of before the end of the week and cut a new release and then get a PR up for version 2.0.0.  I think we will need to add a task to update the response utils to support the prefer response as well.

Hopefully 2.0.0 will be much easier to manage and add features since one of the things we are adding is better testing.

**Comment by Robert-W** [comment]
Getting ready to merge 2.0.0, updating the PR to point to the release branch we will release this under, 1.4.1

File: src/server/utils/response.utils.js [hunk 1]
```diff
@@ -3,6 +3,24 @@ const { VERSIONS } = require('../../constants');
 const errors = require('./error.utils');
 const path = require('path');
 
+const defaultOperation = {
```
Comment (sshah-asymmetrik) [inline src/server/utils/response.utils.js]: Could you make this a function that takes in optional 'id' and 'message' arguments and returns an operation outcome? This way others could supply their own ids/text to suit their needs. If not supplied, these options can default to allok.

File: src/server/utils/response.utils.js [hunk 2]
```diff
@@ -141,6 +159,86 @@ let handleExpandReadResponse = (res, base_version, Resource, resource_json = [])
...
+	resource_version,
+	resourceUrl,
+	status = 201,
+	date,
+	prefer,
+	operationOutcome,
+}) => {
+	const acceptedPreferTypes = ['return=minimal', 'return=representation', 'return=OperationOutcome'];
+
+	switch (prefer) {
```
Comment (Robert-W) [inline src/server/utils/response.utils.js]: Could you pull some of this logic out of the switch statement to shorten this code up a bit? It looks like the following code is in each case and could just be defined above it so it does not need to be redefined.

```javascript
if (resource_version) {
  res.set('Content-Location', `${path.join(resourceUrl, base_version, type, id, '_history', resource_version)}`);
  res.set('ETag', `W/"${resource_version}"`);
}

if (date) {
  res.set('Last-Modified', date.toISOString());
}
res.set('Location', `${base_version}/${type}/${id}`);
```

File: src/server/utils/response.utils.js (hunk 1 above)
Comment (jjosef) [reply src/server/utils/response.utils.js]: Will do asap

File: src/server/utils/response.utils.js (hunk 2 above)
Comment (jjosef) [reply src/server/utils/response.utils.js]: Will do asap

File: src/server/utils/response.utils.js (hunk 1 above)
Comment (jjosef) [reply src/server/utils/response.utils.js]: So I can't figure any way to get the OperationOutcome to produce any useful information based on whats available from the controller. I've created a function that uses these defaults. Perhaps that can be used in the next version somehow.

**Comment by jjosef** [comment]
I've made changes as requested. Was not able to make the generateOperationOutcome method useful without altering controllers significantly, but it is there.

**Comment by Robert-W** [comment]
Thanks, looks good to me. Thanks for contributing @jjosef

**Comment by jjosef** [comment]
No problem, are you going to release a new version on npm? 1.4.2?

**Comment by Robert-W** [comment]
Yup, just did, adding release notes now

**Comment by Robert-W** [comment]
@jjosef Just an FYI, 2.0.0 will be released first thing tomorrow morning. Since it has major changes (some breaking), it won't have this feature in it, and you may want to pin your current server to 1.4.2. I'll also need to publish an update to 2.0.0 to add this feature so it works the same way it does now.

I think we will need to edit these functions, https://github.com/Asymmetrik/phx-tools/blob/master/packages/fhir-response-util/index.js#L60, https://github.com/Asymmetrik/phx-tools/blob/master/packages/fhir-response-util/index.js#L83, to pass back JSON if present. This way, the developer can return any operation outcome or other content based on the header.

**Comment by jjosef** [comment]
Ok thanks so much. If you want any help adding this to 2.0.x as well just let me know (if possible)

**Pull request merged**